src/model/mod.rs

```diff
@@ -21,24 +22,26 @@ pub struct Block {
     pub timestamp: u64,
     /// array of transactions, which successfully passed validation and consensus step.
     pub transactions: Vec<Transaction>,
-    /// hash of a previous block in the chain.
-    //TODO[@humb1t:RH2-9]: what to do if this block first?
-    pub previous_block_hash: Hash,
+    /// Hash of a previous block in the chain.
+    /// Is None only for the first block.
+    pub previous_block_hash: Option<Hash>,
     /// rejected transactions hashes — array of transaction hashes, which did not pass stateful
     /// validation step; this field is optional.
     pub rejected_transactions_hashes: Option<Vec<Hash>>,
 }
 
 impl Block {
     pub fn hash(&self) -> Hash {
```

Comment:
Hash is an array of a constant size (in case of Blake2 it is 32bytes). Can we use fixed length array here?
It is important for scale serialization. If array is fixed length we don't need to store information about the length

Reply:
Yes, it's currently an array of fixed size:
```rust
pub struct Hash(GenericArray<u8, U64>)
```
GenericArray of length 64, that's what Blake2b returns.